### ISSUE TITLE
Polygonal and TriangularDoor have the same value

### ISSUE THREAD

**Issue by dmac000** [NREL/EnergyPlus#5507]
This seems to affect the mapping from `Shape` to `ShapeCat`. Surfaces with more than 4 sides are being mapped to `ShapeCat::Triangular`. From DataSurfaces.cc:

```
int const Triangle( 1 );
int const Quadrilateral( 2 );
int const Rectangle( 3 );
int const Polygonal( 9 );
int const RectangularDoorWindow( 4 );
int const RectangularOverhang( 5 );
int const RectangularLeftFin( 6 );
int const RectangularRightFin( 7 );
int const TriangularWindow( 8 );
int const TriangularDoor( 9 );
```

**Comment by mjwitte** [comment]
@DeadParrot It appears that this could be causing [SurfaceData::computed_shapeCat](https://github.com/NREL/EnergyPlus/blame/b37f69d17bfd8e2075b7125d0448f336ad0200c9/src/EnergyPlus/DataSurfaces.cc#L689-L714) to return the wrong value for polygons.

**Comment by DeadParrot** [comment]
@mjwitte OK, so it looks like fixing the number for Polygonal will give correct results from `computed_shapeCat`. Is there more to it?

How about a changing to an enum (with no specific numbers assigned) while fixing this?

**Comment by mjwitte** [comment]
@DeadParrot That seems reasonable, but I'm puzzled why this didn't introduce diffs when computed_shapeCat was added.

**Comment by DeadParrot** [comment]
@mjwitte Maybe no daylighting or solar reflection test cases have 5+-gons? ShapeCat is only used in PierceSurface.

**Comment by mjwitte** [comment]
That may be true - based on a spot check of existing test files with daylighting they generally have rectangular or 4-sided polygons.  Flr_Rf_8Sides.idf would be a simple candidate with polygon roof/floor to add daylighting to it.

**Comment by mjwitte** [comment]
Never mind - Quadrilateral, duh.

**Comment by mjwitte** [comment]
@DeadParrot It took test file Flr_Rf_8Sides and added daylighting.  Ran with v8.4.0 and with v8.5 develop as of yesterday and got exactly the same daylighting outputs.  Here's a picture of this building.
![image](https://cloud.githubusercontent.com/assets/4991075/13300837/8e9f2876-db07-11e5-943e-92ba2295e390.png)

**Comment by DeadParrot** [comment]
@mjwitte So comparing the different impacts of the `Polygonal( 9 )` bug on `PierceSurface` between v8.4 and v8.5 is complicated (and maybe not worth the effort?):
- In 8.4 there was no special case for triangles so the floor and roof would be treated in the general polygon case, but that case was broken for nonconvex surfaces like these.
- In 8.5 the incorrect `Triangular` `ShapeCat` would make the floor and roof also give incorrect "hit" results.
- Not sure the floor or roof will play any role in daylighting: the `PierceSurface` calls that come to mind are about obstructions between the window and the illumination reference point.

We now have `PierceSurface` unit tests for a range of surface types so if the `Shape` bug is fixed then `ShapeCat` will be correct and we should be test-covered. If you feel an additional surface shape is needed as a test that can be added. Although it is an interesting curiosity, I'm not sure we need to craft a "smoking gun" test case to see the bug in action.

**Comment by mjwitte** [comment]
Sounds reasonable.

**Comment by mjwitte** [comment]
Closed via #5543
